engine/src/main/scala/cromwell/engine/workflow/lifecycle/deletion/DeleteWorkflowFilesActor.scala

```diff
@@ -100,22 +186,48 @@ object DeleteWorkflowFilesActor {
   // Commands
   sealed trait DeleteWorkflowFilesActorMessage
   object StartWorkflowFilesDeletion extends DeleteWorkflowFilesActorMessage
+  object DeleteFiles extends DeleteWorkflowFilesActorMessage
 
   // Actor states
   sealed trait DeleteWorkflowFilesActorState
   object Pending extends DeleteWorkflowFilesActorState
-  object DeletingIntermediateFiles extends DeleteWorkflowFilesActorState
+  object DeleteIntermediateFiles extends DeleteWorkflowFilesActorState
+  object WaitingForIoResponses extends DeleteWorkflowFilesActorState
 
   // State data
   sealed trait DeleteWorkflowFilesActorStateData
   object NoData extends DeleteWorkflowFilesActorStateData
   case class DeletingIntermediateFilesData(intermediateFiles: Set[Path]) extends DeleteWorkflowFilesActorStateData
 
+  case class WaitingForIoResponsesData(commandsToWaitFor: Set[IoDeleteCommand], deleteErrors: List[Throwable] = List.empty, filesNotFound: List[Path] = List.empty) extends DeleteWorkflowFilesActorStateData {
+
+    def deleteCommandComplete(command: IoDeleteCommand): (WaitingForIoResponsesData, DeleteFilesState) = {
+      if (commandsToWaitFor.isEmpty) (this, AllDeleteCommandsDone)
+      else {
+        val updatedCommandsSet = commandsToWaitFor - command
```

Comment:
Would it be a good idea to add a sanity check for the case we hit while mob-programming this – i.e. the size of `commandsToWaitFor` should decrease by one during this operation.
```
assert(commandsToWaitFor.size == updatedCommandsSet - 1)
```

Reply:
As long as you don't use `assert` 😉 

[`Predef.assert`](https://www.scala-lang.org/api/current/scala/Predef$.html) throws a `java.lang.Error` that means "shut down the JVM, now!". We shouldn't stop other workflows from running if this bug occurs. (though it should still be fixed asap)

I'm ok with `require` or some other `java.lang.Exception`.

Reply:
I have added a `require` for sanity check.

Reply:
Right, intended `assert` as pseudo code not the actual impl